### ISSUE TITLE
Sieve Flags and editheader

### ISSUE THREAD

**Issue by anminosti** [Uberspace/manual#309]
Hej,

I've tried to get sieve to set flags and also change the header of my mail. Both of them are currently not working. I can change the location of my mail to another folder.

> require ["fileinto", "reject", "comparator-i;ascii-numeric", "relational", "imap4flags", "variables", "editheader"];
> 
> # Sort mails from mailinglist into folder mailinglist
> if address :is "from" ["test@example.com"]
> {
>     addflag "\\Flagged";
<     addflag "flagvar" "$label3"; #label from thunderbird (it exsits in the dovecot-keywords file)
>     addheader "X-Sieve" "x";
>     stop;
> }

The sieve test is stating that everything is working correctly, unfortunately nothing happens. If I set the flags with Mail applications, I can see them on the server in the file name of the mail.

My suspicion is, that the libaries for "imap4flags" and "editheader" are not enabled. I can't check. Maybe something else is wrong.

Can you please look into the issue and if it is an error on my side, can you please highlight the use of flagging and header editing in the manual. If it is not working for another reason, can you please cover it, so that sieve can also be used for flagging.

Thanks,
anmo

**Comment by brutus** [comment]
Hi anmo,

`imap4flags` is enabled, `editheader` not; current server capabilities are:

```
fileinto reject envelope encoded-character vacation subaddress comparator-i;ascii-numeric relational regex imap4flags copy include variables body enotify environment mailbox date index ihave duplicate mime foreverypart extracttext
```

I just ran a quick test and it works from me with a script like this (without the added header though):

```sieve
require "fileinto";
require "mailbox";
require "imap4flags";

if header :contains "Subject" "flagtest"{
  setflag "\\Seen";
  fileinto :create "Testmails";
}
```

**Comment by brutus** [comment]
I get back to you here about the future status of `editheader` capability soon. (Note though: since we currently in the process of rolling out a new version, it might take a some time, until it's rolled out on a server near you).

**Comment by anminosti** [comment]
Hej brutus,

thanks for your information which capabilities are enabled and also the test run. I managed now to set flags and I'm about it. Even the ones for thunderbird are working.

I'm looking forward to the <code>editheader</code>. Until then I know that it is not implemented.

best anmo

**Comment by brutus** [comment]
Just a quick follow up: the `editheader` plugin will roll out with the next release.

**Comment by noave** [comment]
has been fixed in 7.11.3
